misk/src/main/kotlin/misk/web/extractors/QueryStringParameterExtractor.kt

```diff
@@ -0,0 +1,38 @@
+package misk.web.extractors
+
+import misk.web.QueryParam
+import misk.web.PathPattern
+import misk.web.Request
+import misk.web.actions.WebAction
+import java.util.regex.Matcher
+import kotlin.reflect.KFunction
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.findAnnotation
+
+/**
+ * Creates a [ParameterExtractor] that extracts the Query String parameter with the same name as
+ * [parameter] and returns it as whatever type is specified in [parameter]. Returns null if the
+ * parameter doesn't occur in the request, or can't be parsed into the correct type.
+ */
+object QueryStringParameterExtractorFactory : ParameterExtractor.Factory {
+    override fun create(
+            function: KFunction<*>,
+            parameter: KParameter,
+            pathPattern: PathPattern
+    ): ParameterExtractor? {
+        val pathParamAnnotation = parameter.findAnnotation<QueryParam>() ?: return null
```

Comment:
`queryParamAnnotation` not `pathParamAnnotation`

Reply:
I wrote all this code from scratch